js/markdown/shared/utils.ts

```diff
@@ -128,28 +127,26 @@ export function create_marked({
 	);
 
 	if (header_links) {
-		if (header_links) {
-			marked.use(gfmHeadingId());
-			marked.use({
-				extensions: [
-					{
-						name: "heading",
-						level: "block",
-						renderer(token) {
-							const raw = token.raw
-								.toLowerCase()
-								.trim()
-								.replace(/<[!\/a-z].*?>/gi, "");
-							const id = "h" + slugger.slug(raw);
-							const level = token.depth;
-							const text = this.parser.parseInline(token.tokens!);
-
-							return `<h${level} id="${id}"><a class="md-header-anchor" href="#${id}">${LINK_ICON_CODE}</a>${text}</h${level}>\n`;
-						}
+		marked.use(gfmHeadingId());
+		marked.use({
+			extensions: [
+				{
+					name: "heading",
+					level: "block",
+					renderer(token) {
+						const raw = token.raw
+							.toLowerCase()
+							.trim()
+							.replace(/<[!\/a-z].*?>/gi, "");
+						const id = "h" + slugger.slug(raw);
+						const level = token.depth;
+						const text = this.parser.parseInline(token.tokens!);
+
+						return `<h${level} id="${id}"><a class="md-header-anchor" href="#${id}">${LINK_ICON_CODE}</a>${text}</h${level}>\n`;
 					}
-				]
-			});
-		}
+				}
```

Comment:
since these plugins were being applied _again_ for every marked instance, the highlight code was getting passed through another highlighter. I imagine if we added more instances we would get increasingly weird results.